conans/client/manager.py

```diff
@@ -127,24 +124,24 @@ def download(self, reference, package_ids, remote=None):
         @param remote: install only from that remote
         """
         assert(isinstance(reference, ConanFileReference))
-        remote_proxy = ConanProxy(self._paths, self._user_io, self._remote_manager, remote)
+        remote_proxy = ConanProxy(self._client_cache, self._user_io, self._remote_manager, remote)
 
         if package_ids:
             remote_proxy.download_packages(reference, package_ids)
         else:  # Not specified packages, download all
-            info = remote_proxy.search(str(reference), ignorecase=False)
-            if reference not in info:
+            packages_props = remote_proxy.search_packages(reference, "") # No filter by properties
```

Comment:
Check if second parameter should be None instead of "". And write a test if not existing.